src/commonTest/kotlin/fr/acinq/lightning/channel/states/SpliceTestsCommon.kt

```diff
@@ -43,106 +108,822 @@ class SpliceTestsCommon : LightningTestSuite() {
+        val (alice2, actionsAlice2) = alice1.process(ChannelCommand.WatchReceived(WatchEventConfirmed(alice.channelId, BITCOIN_FUNDING_DEPTHOK, 100, 0, spliceTx)))
+        assertEquals(alice2.commitments.active.size, 2)
+        assertIs<LocalFundingStatus.ConfirmedFundingTx>(alice2.commitments.latest.localFundingStatus)
+        assertEquals(actionsAlice2.size, 3)
+        actionsAlice2.hasWatchFundingSpent(spliceTx.txid)
+        actionsAlice2.has<ChannelAction.Storage.StoreState>()
+        val spliceLockedAlice = actionsAlice2.hasOutgoingMessage<SpliceLocked>()
+
+        val (bob2, actionsBob2) = bob1.process(ChannelCommand.MessageReceived(spliceLockedAlice))
+        assertEquals(bob2.commitments.active.size, 2)
+        assertEquals(actionsBob2.size, 1)
+        actionsBob2.has<ChannelAction.Storage.StoreState>()
+        val (bob3, actionsBob3) = bob2.process(ChannelCommand.WatchReceived(WatchEventConfirmed(bob.channelId, BITCOIN_FUNDING_DEPTHOK, 100, 0, spliceTx)))
+        assertEquals(bob3.commitments.active.size, 1)
+        assertIs<LocalFundingStatus.ConfirmedFundingTx>(bob3.commitments.latest.localFundingStatus)
+        assertEquals(actionsBob3.size, 3)
+        actionsBob3.hasWatchFundingSpent(spliceTx.txid)
+        actionsBob3.has<ChannelAction.Storage.StoreState>()
+        val spliceLockedBob = actionsBob3.hasOutgoingMessage<SpliceLocked>()
+
+        val (alice3, actionsAlice3) = alice2.process(ChannelCommand.MessageReceived(spliceLockedBob))
+        assertEquals(alice3.commitments.active.size, 1)
+        assertEquals(actionsAlice3.size, 1)
+        actionsAlice3.has<ChannelAction.Storage.StoreState>()
+    }
+
+    @Test
+    fun `exchange splice_locked -- zero-conf`() {
+        val (alice, bob) = reachNormal(zeroConf = true)
+        val (alice1, bob1) = spliceOut(alice, bob, 60_000.sat)
+        val spliceTx = alice1.commitments.latest.localFundingStatus.signedTx!!
+
+        val (alice2, actionsAlice2) = alice1.process(ChannelCommand.MessageReceived(SpliceLocked(alice.channelId, spliceTx.hash)))
+        assertEquals(actionsAlice2.size, 1)
+        actionsAlice2.has<ChannelAction.Storage.StoreState>()
+        assertEquals(alice2.commitments.active.size, 1)
+        assertNotEquals(alice2.commitments.latest.fundingTxId, alice.commitments.latest.fundingTxId)
+        assertIs<LocalFundingStatus.UnconfirmedFundingTx>(alice2.commitments.latest.localFundingStatus)
+
+        val (bob2, actionsBob2) = bob1.process(ChannelCommand.MessageReceived(SpliceLocked(bob.channelId, spliceTx.hash)))
+        assertEquals(actionsBob2.size, 1)
+        actionsBob2.has<ChannelAction.Storage.StoreState>()
+        assertEquals(bob2.commitments.active.size, 1)
+        assertNotEquals(bob2.commitments.latest.fundingTxId, bob.commitments.latest.fundingTxId)
+        assertIs<LocalFundingStatus.UnconfirmedFundingTx>(bob2.commitments.latest.localFundingStatus)
+    }
+
+    @Test
+    fun `use channel before splice_locked -- zero-conf`() {
```

Comment:
The test doesn't pass because this corner case has not been implemented. We did discuss it though! The check below should actually be `commits.size < active.size`, and then we should only consider the N first sigs, N being the number of active commitments:
https://github.com/ACINQ/lightning-kmp/blob/1b376e37571ea74089400886e0999c221ddaaeb6/src/commonMain/kotlin/fr/acinq/lightning/channel/Commitments.kt#L708-L711

Reply:
This was indeed trivial to fix! Done in https://github.com/ACINQ/lightning-kmp/pull/438/commits/71b0841611b8a3004eee0fba397954e52820e0f7